Add Expansion and Rotation control systems to BBH exec

knelli2 opened the pull request:
## Proposed changes

To do this

1. Modify the control system FunctionsOfTimeInitialize tag to allow functions of time to be overridden even if they have control systems
2. If all controlled functions of time are overridden, have the control system trigger return that it's never triggered (so we don't do extra horizon finds)
3. Use the control system FunctionsOfTimeInitialize tag in the DgDomain initialization action if we have control systems.

<!--
At a high level, describe what this PR does.
-->

### Upgrade instructions

<!--
If this PR makes changes that other people should be aware of when upgrading
their code, describe what they should do between the two UPGRADE INSTRUCTIONS
lines below.
-->
<!-- UPGRADE INSTRUCTIONS -->

<!-- UPGRADE INSTRUCTIONS -->

### Code review checklist

- [ ] The code is documented and the documentation renders correctly. Run
  `make doc` to generate the documentation locally into `BUILD_DIR/docs/html`.
  Then open `index.html`.
- [ ] The code follows the stylistic and code quality guidelines listed in the
  [code review guide](https://spectre-code.org/code_review_guide.html).
- [ ] The PR lists upgrade instructions and is labeled `bugfix` or
  `new feature` if appropriate.

### Further comments

<!--
If this is a relatively large or complex change, kick off the discussion by
explaining why you chose the solution you did and what alternatives you
considered, etc...
-->

knelli2 committed:
Allow FoT tag to both read in data and have it set by control systems

knelli2 committed:
Add UseControlSystems template bool to dg initialize domain action

knelli2 committed:
Add Expansion and Rotation control systems to BBH exec

knelli2 committed:
Only trigger control sys event when at least one control sys is active

knelli2 committed:
Remove ControlError tag from Rotation system

This is already taken care of in the Initialization action

markscheel reviewed (commented):
Looks good.  One potential confusion: It took me a while to figure out what `override_functions_of_time` was for (who is overriding them? how are they overridden and why?).

Should `override_functions_of_time` be renamed `read_functions_of_time_from_file`?  And similar with `OverrideCubicFunctionsOfTime` in DgDomain.hpp and there may be another similar name elsewhere.

knelli2 commented:
@markscheel Yes that seems like a reasonable rename. I can do that in a follow-up PR if you'd like?

knelli2 commented:
@wthrowe @markscheel Rebased and fixed conflicts

knelli2 committed:
Specify tensors in FindHorizon submeasurement

If these were left as a template pack, the compiler was unable to
find the correct "apply" function in the db::apply call in the
control system Event.

knelli2 commented:
@wthrowe @markscheel Is this waiting on me or are you both satisfied with the changes?

wthrowe reviewed (approved):
I've been waiting for @markscheel to reply.

markscheel commented:
I am fine with renaming in a separate PR, since this one is so close to merging.
